Jenkinsfile_CNP

```diff
@@ -42,4 +58,45 @@ withPipeline(type, product, component) {
             Using NONPROD_ENVIRONMENT_NAME: ${env.NONPROD_ENVIRONMENT_NAME}""".stripIndent()
   }
 
+  before('functionalTest:idam-preview') {
+    env.PUBLIC_URL = 'https://idam-web-public.preview.platform.hmcts.net'
+    env.TESTING_SUPPORT_API_URL = 'https://idam-testing-support-api.preview.platform.hmcts.net'
+    env.TESTING_SERVICE_CLIENT = 'idam-functional-test-service'
+    println """\
+            Using PREVIEW_ENVIRONMENT_NAME: ${env.PREVIEW_ENVIRONMENT_NAME}
+            Using NONPROD_ENVIRONMENT_NAME: ${env.NONPROD_ENVIRONMENT_NAME}
+            Using PUBLIC_URL: ${env.PUBLIC_URL}""".stripIndent()
+  }
+
+  before('functionalTest:idam-aat') {
+    env.PUBLIC_URL = 'https://idam-web-public.aat.platform.hmcts.net'
+    env.TESTING_SUPPORT_API_URL = 'https://idam-testing-support-api.aat.platform.hmcts.net'
+    env.TESTING_SERVICE_CLIENT = 'idam-functional-test-service'
+    println """\
+                Using NONPROD_ENVIRONMENT_NAME: ${env.NONPROD_ENVIRONMENT_NAME}
+                Using TESTING_SUPPORT_API_URL: ${env.TESTING_SUPPORT_API_URL}""".stripIndent()
+  }
+
+  afterSuccess('functionalTest:idam-preview') {
```

Comment:
Should this be `afterAlways`?
I recently realised that in User Dashboard we only get test reports if all tests pass - maybe that's the reason?